misc/hawkbit-upload.py

```diff
@@ -598,36 +696,43 @@ def ensure_filter(filters, query: str, name: str):
 
     filters = client.get_all_targetfilters().get("content") or []
 
-    boot_filter = ensure_filter(
-        filters,
-        f'attribute.boot_mode == "{args.bootmode}"',
-        f"Boots from {args.bootmode}",
-    )
     channel_filter = ensure_filter(
+        client,
         filters,
         f'attribute.update_channel == "{args.channel}"',
         f"Downloads from {args.channel} channel",
     )
 
-    print(f"Boot filter is {boot_filter}")
     print(f"Channel filter is {channel_filter}")
 
-    print("Creating software module")
-    client.add_softwaremodule(name=args.softwareModule)
+
+    print("Creating or updating software module")
+    client.add_or_update_softwaremodule(name=args.softwareModule)
+    
     print("Creating Distribution set")
-    dist_id = client.add_distributionset(
+    dist_id = client.add_or_update_distributionset(
         args.distribution, module_ids=[client.get_softwaremodule().get("id")]
     )
-    print("uploading artifact")
-    client.add_artifact(args.bundle)
-
-    # print("Creating rollout")
-    # rollout = client.createRollout(
-    #     f"{args.distribution.capitalize()} {args.version}",
-    #     dist_id=dist_id,
-    #     target_filter_query=f"{boot_filter.get("query")} and {channel_filter.get("query")}",
-    #     autostart=True,
-    # )
-    # print(json.dumps(rollout))
+
+    print("Uploading new artifact and removing all existing ones")
+    client.add_or_update_artifact(args.bundle)
+
+    # Create or replace the rollout
+    raucb_filename = os.path.basename(args.bundle)
+    rollout_name = f"Nightly_{raucb_filename}"
```

Comment:
We are passing the channel as a last parameter to the script - no need to hardcode it